src/configuration.ts

```diff
@@ -536,11 +556,31 @@ export function parseConfigFile(
             return {};
         }
         return {
-            exclude: arrayify(raw.exclude).map(
-                (pattern) => dir === undefined ? path.resolve(pattern) : path.resolve(dir, pattern),
-            ),
+            exclude: parsePatterns(raw.exclude, dir),
         };
     }
+
+    function parsePatterns(patterns?: string | string[], dir?: string): string[] {
+        return arrayify(patterns).map((pattern) => (
+            dir === undefined ? path.resolve(pattern) : path.resolve(dir, pattern)
```

Comment:
this does not what you want it to do:
* on windows it uses backslashes. these are typically used as escape characters in glob patterns. `minimatch` has some special handling for backslashes on windows, so this evens out
* negated globs are destroyed: `!*.js` -> `/dir/!*.js` instead of `!/dir/*.js`

there's a package `to-absolute-glob` that handles this well